File: archive/go/2020/aug/distribute-candies.go
package aug

import "math"

func Distribute(candies int, num int) (distributes []int) {
	var res = make([]int, num)
	for i := 0; candies > 0; candies -= i {
		i += 1
		res[i%num] += int(math.Min(float64(candies), float64(i)))
	}
	return res
}

File: archive/python/2020/aug/distribute-candies.py
## distribute candies 1

def distribute(candies):
    ### 利用set去重
    return min(len(candies) / 2, len(set(candies)))


def distributeCandies(candies, num):
    res = [0] * num
    i = 0
    while candies > 0:
        res[i % num] += min(candies, i + 1)
        candies -= i + 1
        i += 1
    return res
